mage_ai/data_cleaner/column_type_detector.py

```diff
@@ -63,18 +63,23 @@ def str_in_set(string, string_set):
     return any(entry in string for entry in string_set)
 
 
-def get_mismatched_rows(series, column_type):
+def find_syntax_errors(series, column_type):
     if len(series) == 0:
-        return []
-    mismatched_rows = []
+        return pd.Series([])
+    mask = pd.Series([False] * len(series))
+    mask.index = series.index
     if column_type == EMAIL:
-        mismatched_rows = series[~series.str.match(REGEX_EMAIL)].tolist()
+        mask |= ~series.str.match(REGEX_EMAIL)
```

Comment:
why do you need mask? Can you just return `~series.str.match(xxx)`

Reply:
Fixed